pkg/provider/vault/vault.go

```diff
@@ -201,8 +204,15 @@ func (v *client) readSecret(ctx context.Context, path, version string) (map[stri
 			byteMap[k] = []byte(t)
 		case []byte:
 			byteMap[k] = t
+		case map[string]interface{}:
+			jsonString, err := json.Marshal(t)
+			byteMap[k] = jsonString
+			if err != nil {
+				return nil, err
+			}
+
 		default:
-			return nil, errors.New(errVaultData)
+			return nil, errors.New(fmt.Sprintf("Secret data not in expected format: %v", secretData))
```

Comment:
Please create a new declared error string on top for this one, just to follow the pattern